src/main/java/games/strategy/engine/framework/ui/NewGameChooserEntry.java

```diff
@@ -37,9 +42,21 @@ private String getLowerCaseComparable(final NewGameChooserEntry newGameChooserEn
     };
   }
 
+  public NewGameChooserEntry() {
+    this(defaultGameUri());
+  }
+
 
-  public NewGameChooserEntry(final URI uri)
-      throws IOException, GameParseException, SAXException, EngineVersionException {
+  private static URI defaultGameUri() {
+    final String defaultLocation = ClientSetting.SELECTED_GAME_LOCATION.value();
+    try {
+      return new URI(defaultLocation);
+    } catch (final URISyntaxException e) {
+      throw new IllegalStateException("Illegal default game location: " + defaultLocation, e);
+    }
```

Comment:
Note that this is basically the same implementation as `URI#create()` (except it throws `IllegalArgumentException` instead of `IllegalStateException`).  So that the entire method body can be replaced with

```java
return URI.create(ClientSetting.SELECTED_GAME_LOCATION.value());
```

and the `URISyntaxException` can be removed.